### ISSUE TITLE
Monitoring - after install rancher-monitoring-kube-scheduler is down

### ISSUE THREAD

**Issue by ronhorton** [rancher/rancher#36876]
**Setup**
- Rancher version: `2.6.4-rc8`
- Browser type & version: `Chrome 98.0.4758.80 (Official Build) (x86_64)`

**Describe the bug**
On the prometheus rules page under monitoring, when I switch from `only user namespaces` to `all namespaces`, a 500 error appears in the console - `Failed to load resource: the server responded with a status of 500 ()`

**To Reproduce**
1. create downstream cluster - ec2 3 w, 1 cp, 1 etcd - `k8s 1.23`
2. install monitoring v2 100.1.1+up19.0.3
3. install alerting drivers 100.0.2
4. click `prometheus targets`

**Result**
[serviceMonitor/cattle-monitoring-system/rancher-monitoring-kube-scheduler/0 (0/1 up)]

**Expected Result**
all targets up and active

**Screenshots**

<img width="1569" alt="Screen Shot 2022-03-14 at 10 49 49 AM" src="https://user-images.githubusercontent.com/89212856/158197944-d09ce07e-d873-4b19-a5ac-89fd288fd4bf.png">

**Additional context**
This doesn't occur in k8s version 1.22 clusters

**Detailed Description**

**Context**

**Comment by ronhorton** [comment]
Reproducing to get logs; on reproducing, `rancher-monitoring-kube-controller-manager has 1/2 up`

Error from pushprox-kube-controller-manager-client: 
`level=error ts=2022-03-14T16:25:02.534Z caller=main.go:101 err="failed to scrape https://127.0.0.1:10257/metrics: Get \"https://127.0.0.1:10257/metrics\": dial tcp 127.0.0.1:10257: connect: connection refused"`

Error from pushprox-kube-scheduler-client:
`level=error ts=2022-03-14T16:27:09.491Z caller=main.go:101 err="failed to scrape http://127.0.0.1:10251/metrics: Get \"http://127.0.0.1:10251/metrics\": dial tcp 127.0.0.1:10251: connect: connection refused"
level=info ts=2022-03-14T16:27:09.492Z caller=main.go:113 msg="Pushed failed scrape response"`

![Screen Shot 2022-03-14 at 12.34.19 PM.png](https://zube.io/files/rancher/fb74f226a6824a44eba2c63b1c52a60c-screen-shot-2022-03-14-at-12-34-19-pm.png)

**Comment by aiyengar2** [comment]
Root cause is this: https://github.com/rancher/charts/pull/1766#discussion_r813497643

**Comment by aiyengar2** [comment]
@ronhorton when testing this fix, we should double check that both k8s 1.23 clusters work as expected (reproducing the original issue and verifying the fix) **and** that k8s <1.22 clusters see no changes, since the fix I'm putting in for this could affect both.

**Comment by aiyengar2** [comment]
Cannot move forward with this issue or verifying the fix in my PR until RKE1 resolves https://github.com/rancher/rancher/issues/36899.

cc: @MKlimuszka @SheilaghM

**Comment by aiyengar2** [comment]
We'll also need to add a release-note that Monitoring an RKE1 cluster that is >1.23.0-0 but less than the version that contains the fix that resolves https://github.com/rancher/rancher/issues/36899 will be broken.

e.g. `v1.23.4-rancher1-1` will never have a fix.

cc: @kinarashah do you agree?

**Comment by kinarashah** [comment]
@aiyengar2 you're right from technical point of view but we haven't released `v1.23` yet. `v1.23.4` is the first 1.23.x version going to be available and `v1.23.4-rancher1-1` is currently only available via `dev-v2.6` KDM branch and RCs. So I don't think we need to release note this, tho if we announced that secure serving will work in kube-scheduler for 1.22 - it would be good to call it out as a bug.

**Comment by aiyengar2** [comment]
That's a good call-out Kinara. I don't think we release-noted that secure serving will work in 1.22 before, so I'll remove the release note from both of these issues.

**Comment by aiyengar2** [comment]
This will need to be tested in a release after [v2.6.4-rc10](https://github.com/rancher/rancher/releases/tag/v2.6.4-rc10), since that's when the RKE1 fixes were vendored in.
